### PR TITLE
AG-9588 Add validation for user theme

### PR DESCRIPTION
**Pull request by olegat**
This isn't an ideal solution, but it's an improvement to just naively converting user data to a theme with `as AgChartTheme` with no validation whatsoever.

In long term, we should find some way to integrate this with the _validatation.ts_ utility.

This will solve the bug where nothing is rendered, but it’s not an ideal solution. Unfortunately it seems like user themes are not integrated with our _validation.ts_ utility (which would be an ideal solution).

This PR implements some rudimentary validation to try to prevent unexpected behaviours like no rendering. However, as a consequence, the entire `options.theme` value is thrown away if an error is found. Migrating to the _validation.ts_ will enhance our ability to control the defaults when validation fails.

I think that the current state of the solution is acceptable; nevertheless, it is tech debt. I don’t believe that making further enhancements to this solution is a good idea.

### CODE REVIEW

File: packages/ag-charts-community/src/chart/mapping/themes.ts
Comment (olegat) [inline packages/ag-charts-community/src/chart/mapping/themes.ts:59]: Note: this expression is no longer necessary because the two if-statements above remove `ChartTheme | undefined | string` from the union, leaving `AgChartTheme` the only remaining type.

File: packages/ag-charts-community/src/chart/mapping/themes.ts
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/mapping/themes.ts:68]: Nit: The `typeof palette === 'object'` check makes this guard redundant?

File: packages/ag-charts-community/src/chart/mapping/themes.ts
Comment (alantreadway) [inline packages/ag-charts-community/src/chart/mapping/themes.ts:59]: We aren't handling the `typeof baseTheme === 'object'` case 100% correctly here, but I think on balance that is fine since strictly the recursive handling is an undocumented feature.

**Review by alantreadway — APPROVED** [review]
LGTM, please see if you can address the single nit before merge?

File: packages/ag-charts-community/src/chart/mapping/themes.ts
Comment (olegat) [reply packages/ag-charts-community/src/chart/mapping/themes.ts:68]: You'd think so, but my `tsserver` tells me that the type of `palette` is `object | null` after the first guard.

This might be okay, but I don't know how the `const { fills, strokes } = palette` deconstruction will behave if the palette is `null`.

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**
